### ISSUE TITLE
torch.where should mention Bool rather than Byte when given wrong dtype mask

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#49769]
## 🐛 Bug

![image](https://user-images.githubusercontent.com/5674597/102943344-bb617e80-4485-11eb-8dcc-2124bb6a7314.png)

 - PyTorch Version (e.g., 1.0): 1.7.0

**Comment by ngimel** [comment]
Thanks for reporting!

**Comment by saiamrit** [comment]
Hi @ngimel @mruberry , I would like to work on this issue, kindly guide me how to start with this.

**Comment by gchanan** [comment]
@saiamrit: you can find this code here: https://github.com/pytorch/pytorch/blob/963f7629b591dc9750476faf1513bc7f1fb4d6de/aten/src/ATen/native/TensorCompare.cpp#L217-L219

For general contributing tips: https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md

**Comment by jonykarki** [comment]
@saiamrit Are you working on this? If not, I'd like to work on this.

**Comment by mruberry** [comment]
@jonykarki Go ahead, as the contributing guide mentions we don't support "reserving" issues.

**Comment by jonykarki** [comment]
Should I just replace `Byte` with `Bool` or make it `Byte or Bool` since having `Byte` condition works as well?
![image](https://user-images.githubusercontent.com/25265687/104137986-8e0e3f00-5366-11eb-8c78-d1bc5abb83b4.png)

**Comment by ngimel** [comment]
Replace `Byte` with `Bool`, we should deprecate and disallow usage of `Byte` here completely.
